ToolChain/Assembler/Parser/Parser.py

```diff
@@ -20,586 +19,395 @@
+            # Form is invalid, doesn't fit any state descriptions
+            raise ValueError("Invalid instruction format")
 
-        return buildInstruction, lineInstruction
+        return state
 
-    def _evaluateAndExtractInfoFromLine(self, line, buildInstruction, rawLine):
+    def _evaluateIndicatorData(self, text, dataIdentifier, line, instruction):
         """
-        This is a helper method. The aim to this is to keep caller code cleaner
-        Method evaluates each part of the line to determine whether we're dealing with an instruction, label, data, etc.
-        We begin by verifying if the first argument has label or data mnemonics (not an instruction).
-        If none are found, we assume it's an instruction. We then proceed to evaluating the arguments.
-        The method uses the Instruction class' source and destination fields to enforce proper instruction format.
-        For example, if the source and destination have already been found, additional arguments means the line is an
-        invalid instruction.
-        :param line: the line we want to parse, already split into individual arguments
-        :param buildInstruction: Our template for the instruction, used to later build the instruction
-        :param rawLine: the line as we read it originally, preserved so we don't consume extra spaces for .dataAlpha
-        :return:
+        In this method, we are dealing with identifiers, meaning either .dataAlpha, .dataNumeric, .global, a label, or
+        a comment.
+        :param text: str, The raw line of text as read from the file directly
+        :param dataIdentifier: str, The first token of the line, used to determine which identifier we have
+        :param line: str list, The line split into individual tokens using a space delimiter by default
+        :param instruction: bytestring, Will contain the piece of data based on the appropriate identifier
+        :return: bytestring, Instruction containing relevant data, and the appropriate flag to be used by the assembler
         """
 
+        labelFlag = INSTRUCTION_FLAG
+
+        # This ensures that if we have a memory reference, that it is not part of a comment.
+        # Ex: ";label:" should not enter this condition, but "label:" should.
+        if dataIdentifier[-1] == MEMORY_REFERENCE_INDICATORS and dataIdentifier[0] != COMMENT_INDICATORS:
+            if dataIdentifier.count(MEMORY_REFERENCE_INDICATORS) > 1:
+                # Forcing coding standard: labels can't contain colons ":"
+                raise ValueError("Syntax error, memory reference has too many \":\"")
+            instruction = dataIdentifier[:-1].upper()
+            labelFlag = LOCAL_REFERENCE_FLAG
+
+        elif dataIdentifier == EXPORTED_REFERENCE_INDICATOR:
+            # Global (external) label that can be used by other files
+            instruction = line[1].upper()
+            labelFlag = GLOBAL_REFERENCE_FLAG
+
+        elif dataIdentifier == DATA_ALPHA_INDICATOR:
```

Comment:
Since the changes I'm proposing avoids splitting the operands early, (micro optimization and cleaning up the style),  it requires that in this function, we split ourselves for all cases BUT `.dataAlpha`. So this is how we tidy up this function to match the rest:
```python3
if mnemonic == DATA_ALPHA_INDICATOR:
   # same code except replace text with operand
  return # needed information
operands = operands.upper().split()

if dataIdentifier[-1] == MEMORY_REFERENCE_INDICATORS and dataIdentifier[0] != COMMENT_INDICATORS:
   # ...
else: 
   raise # Err
return # needed information
```